src/neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -26,7 +26,10 @@ public bool CheckTransaction(Transaction tx, StoreView snapshot)
             BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
             senderFee.TryGetValue(tx.Sender, out var totalSenderFeeFromPool);
             BigInteger fee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
-            return balance >= fee;
+            if (balance >= fee) return true;
+
+            BigInteger unclaimed = NativeContract.NEO.UnclaimedGas(snapshot, tx.Sender, snapshot.Height);
```

Comment:
This is in the hot path for transaction verification and after #1848 this call can be quite expensive (with a number of DB accesses).

Reply:
But I think this is necessary anyway @roman-khimov , because we must know if someone has enough funds for transfer... it really costs to calculate, but we don't need precise value. Maybe, if we have some caching on unclaimedGas values, we could call some method `EnoughUnclaimedGas(VALUE)`, so that if caching already has enough, we don't recalculate. Truth is, we just need to know if Unclaimed is bigger than VALUE, not its precise value.

Reply:
If it has funds, there will be no problem, otherwise, we need to check it.